### PR TITLE
Correct guest info guest id in results.log and guest_info.json

### PR DESCRIPTION
**Pull request by keirazhang**
With latest VMTools, the guest id in guest info for MIRACL LINUX 9.2 is empty string, but in our results.log it is not. By checking log plugin, it erased all empty values. So this fix is to correct that. All facts will take the last value as its final value, no matter it is empty or not.

```
+------------------------------------------------------------------------------------------------------------+
| Config Guest Id           | asianux8_64Guest                                                               |
+------------------------------------------------------------------------------------------------------------+
| GuestInfo Guest Id        |                                                                                |
+------------------------------------------------------------------------------------------------------------+
| GuestInfo Guest Full Name | Linux 5.14.0-284.11.1.el9_2.x86_64 MIRACLE LINUX 9.2 (Feige) 9.2 MIRACLE LINUX |
|                           | 9.2 (Feige) cpe:/o:cybertrust_japan:miracle_linux:9                            |
+------------------------------------------------------------------------------------------------------------+
| GuestInfo Guest Family    | linuxGuest                                                                     |
+------------------------------------------------------------------------------------------------------------+
| GuestInfo Detailed Data   | architecture='X86'                                                             |
|                           | bitness='64'                                                                   |
|                           | cpeString='cpe:/o:cybertrust_japan:miracle_linux:9'                            |
|                           | distroAddlVersion='9.2 (Feige)'                                                |
|                           | distroName='MIRACLE LINUX'                                                     |
|                           | distroVersion='9.2'                                                            |
|                           | familyName='Linux'                                                             |
|                           | kernelVersion='5.14.0-284.11.1.el9_2.x86_64'                                   |
|                           | prettyName='MIRACLE LINUX 9.2 (Feige)'                                         |
+------------------------------------------------------------------------------------------------------------+

Test Results (Total: 1, Passed: 1, Elapsed Time: 00:06:40)
+--------------------------------------------------+
| ID | Name                   | Status | Exec Time |
+--------------------------------------------------+
|  1 | ovt_verify_pkg_install | Passed | 00:06:20  |
+--------------------------------------------------+
```

### CODE REVIEW

File: plugin/ansible_vsphere_gosv_log.py [hunk 1]
```diff
@@ -930,26 +930,24 @@ def _collect_ansible_gosv_facts(self, result):
...
-                    if guestinfo_hash not in self.collected_guest_info:
-                        self.collected_guest_info[guestinfo_hash] = vm_guest_info
+                    self.test_runs[self._last_test_id].name = ansible_facts['current_testcase_name']
+
+                if task_file == 'vm_get_guest_info.yml':
+                    esxi_build = self._ansible_gosv_facts.get('esxi_build','')
+                    vm_hw_version = self._ansible_gosv_facts.get('vm_hardware_version','')
+                    vmtools_version = self._ansible_gosv_facts.get('guestinfo_vmtools_info','')
+                    if (esxi_build and vm_hw_version and vmtools_version):
+                        guestinfo_hash = str(hash("{}{}{}".format(esxi_build, vm_hw_version, vmtools_version)))
```
Comment (Tomorrow9) [inline plugin/ansible_vsphere_gosv_log.py:946]: In the same test run, is this 'esxi_build' the same?

Comment (keirazhang) [reply plugin/ansible_vsphere_gosv_log.py:946]: Yes, but we need to consider special scenario that user may define a customized playbook or test case to run collect guest info on different ESXi server, with different hardware version and different VMTools.

Comment (Tomorrow9) [reply plugin/ansible_vsphere_gosv_log.py:946]: Ok, it's no big deal.

**Review by Tomorrow9 — APPROVED** [review]
LGTM, thanks

**Pull request merged**